### ISSUE TITLE
Pick-a-date should have improved keyboard navigation

### ISSUE THREAD

**Issue by seanupton** [upiq/uu.task#28]
Ideally pick-a-date pattern in mockup should have some UI improvements to make it easier to use by keyboard:
- Escape key should act same as delete/backspace.
- Time: possible to type time directly, with auto-scroll to appropriate slot, if applicable, since there are only two possible l10n scenarios: 24h and 12h (am/pm).  Should be possible to type H:M pattern, (optionally followed by 'A' or 'P' in 12 hour locales).
- Ideally parse basic natural typed input with locale support, similarly to https://teamspace.upiq.org/static/demo/smartdate/demo.html or https://teamspace.upiq.org/static/demo/smartdate/demo_fr.html ([uu.smartdate](https://github.com/upiq/uu.smartdate) uses https://github.com/datejs/Datejs for this, but there may be alternatives without such dependency).
